In [31]:
#Importing The Dependency
import pandas as pd
import numpy as np

In [8]:
#Reading The Dataset
df=pd.read_csv("/content/netflix_titles_nov_2019.csv.zip")
df.head()

show_id                                title                    director  \
0  81193313                            Chocolate                         NaN   
1  81197050  Guatemala: Heart of the Mayan World  Luis Ara, Ignacio Jaunsolo   
2  81213894                      The Zoya Factor             Abhishek Sharma   
3  81082007                            Atlantics                   Mati Diop   
4  80213643                      Chip and Potato                         NaN   

                                                cast  \
0  Ha Ji-won, Yoon Kye-sang, Jang Seung-jo, Kang ...   
1                                  Christian Morales   
2  Sonam Kapoor, Dulquer Salmaan, Sanjay Kapoor, ...   
3  Mama Sane, Amadou Mbow, Ibrahima Traore, Nicol...   
4  Abigail Oliver, Andrea Libman, Briana Buckmast...   

                    country         date_added  release_year rating  \
0               South Korea  November 30, 2019          2019  TV-14   
1                       NaN  November 30, 2019          2019   TV-G   
2                     India  November 30, 2019          2019  TV-14   
3  France, Senegal, Belgium  November 29, 2019          2019  TV-14   
4    Canada, United Kingdom                NaN          2019   TV-Y   

    duration                                          listed_in  \
0   1 Season  International TV Shows, Korean TV Shows, Roman...   
1     67 min                Documentaries, International Movies   
2    135 min             Comedies, Dramas, International Movies   
3    106 min   Dramas, Independent Movies, International Movies   
4  2 Seasons                                           Kids' TV   

                                         description     type  
0  Brought together by meaningful meals in the pa...  TV Show  
1  From Sierra de las Minas to Esquipulas, explor...    Movie  
2  A goofy copywriter unwittingly convinces the I...    Movie  
3  Arranged to marry a rich man, young Ada is cru...    Movie  
4  Lovable pug Chip starts kindergarten, makes ne...  TV Show

In [67]:
df.shape

(5837, 13)

There are 5837 Rows and 13 Columns in are Dataset

In [9]:
#Checking The Null Values
pd.isnull(df).sum()

show_id            0
title              0
director        1901
cast             556
country          427
date_added       642
release_year       0
rating            10
duration           0
listed_in          0
description        0
type               0
dtype: int64

In [14]:
# Checking The percentage of missing values in each column
missing_value_percentage= pd.isnull(df).mean()*100
missing_value_percentage

show_id          0.000000
title            0.000000
director        32.568100
cast             9.525441
country          7.315402
date_added      10.998801
release_year     0.000000
rating           0.171321
duration         0.000000
listed_in        0.000000
description      0.000000
type             0.000000
dtype: float64

The above code shows the percentage of missing values in each column. The director column has the highest percentage of missing values i.e 32.5% followed by date_added(9.5%) and country(7.4%).


In [20]:
# Checking total number of rows with missing data
print(df.isnull().any(axis=1).sum())

2390


There are 2390 Rows that have some missing data

In [22]:
# Handling missing values
df['director'] = df['director'].fillna("Unknown")
df['cast'] = df['cast'].fillna("Not Available")
df['country'] = df['country'].fillna('Not Available')
df['date_added_missing'] = df['date_added'].isnull().astype(int)
df['date_added'] = df['date_added'].fillna("Unknown")
df['rating'] = df['rating'].fillna(df['rating'].mode()[0])


I have filled the missing values such that the information of the show is not get incorrect. as i fill all the categorical values with a new category 'Unknown' or 'Not Available'. if i has used the mode for filling missing values it have a great chance to get incorrect information for a show.

In [32]:
# Checking for duplicate values
df.duplicated().sum()

np.int64(0)

The Data has no duplicate Rows

In [60]:
#Convert date formats to a consistent type (e.g., dd-mm-yyyy)
df['date_added'] = pd.to_datetime(df['date_added'], errors='coerce', dayfirst=True)

In [66]:
df['date_added'].head()

0   2019-11-30
1   2019-11-30
2   2019-11-30
3   2019-11-29
4          NaT
Name: date_added, dtype: datetime64[ns]

The date has been changed to dd-mm-yyyy type with missing dates represented by NaN

In [63]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5837 entries, 0 to 5836
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   show_id             5837 non-null   int64         
 1   title               5837 non-null   object        
 2   director            5837 non-null   object        
 3   cast                5837 non-null   object        
 4   country             5837 non-null   object        
 5   date_added          5195 non-null   datetime64[ns]
 6   release_year        5837 non-null   int64         
 7   rating              5837 non-null   object        
 8   duration            5837 non-null   object        
 9   listed_in           5837 non-null   object        
 10  description         5837 non-null   object        
 11  type                5837 non-null   object        
 12  date_added_missing  5837 non-null   int64         
dtypes: datetime64[ns](1), int64(3), object(9)
memory

All the datatypes is in proper formate

This was only the initial data cleaning stage. Following this, we can perform Exploratory Data Analysis (EDA) for graphical insights and proceed with encoding categorical variables as part of the preprocessing required for fitting a machine learning model